deployments/databricks/deploy_collector.sh

```diff
@@ -0,0 +1,170 @@
+
+if [ $OTEL_VERSION = "latest" ]; then
+        OTEL_VERSION=$(curl --silent "$OTEL_API_URL" |    # Get latest Collector release from GitHub api
+            grep '"tag_name":' |                          # Get tag name line
+            sed -E 's/.*"([^"]+)".*/\1/')                 # Pluck latest release version
+        if [ -z "$OTEL_VERSION" ]; then
+            echo "Failed to get tag_name for latest release from $OTEL_VERSION/latest" >&2
+            exit 1
+        fi
+fi
+
+OTEL_BINARY_DOWNLOAD_URL=${OTEL_DOWNLOAD_BASE_URL}/download/${OTEL_VERSION}/otelcol_${OS}
+
+# Download Splunk's distribution of the OpenTelemetry Collector
+mkdir -p "$SCRIPT_DIR"
+wget -nv -O "$OTEL_BINARY_FILE" $OTEL_BINARY_DOWNLOAD_URL
+chmod +x "$OTEL_BINARY_FILE"
+
+# The Spark receiver should only be run in one instance per-Cluster. Run
+# it on the driver node, as there's one per-cluster.
+# More info on Databricks init script environment variables:
+# https://docs.databricks.com/en/init-scripts/environment-variables.html#use-secrets-in-init-scripts
+if [ $DB_IS_DRIVER = "TRUE" ]; then
+        OPTIONAL_SPARK_RECEIVER=", apachespark"
+else
+        OPTIONAL_SPARK_RECEIVER=""
+fi
+
+collector_config="
+extensions:
+  bearertokenauth:
+    token: "$DATABRICKS_ACCESS_TOKEN"
+
+receivers:
+  apachespark:
+    # https://community.databricks.com/t5/data-engineering/how-to-obtain-the-server-url-for-using-spark-s-rest-api/td-p/83410
+    endpoint: https://$DATABRICKS_CLUSTER_HOSTNAME/driver-proxy-api/o/0/$DB_CLUSTER_ID/40001
+    auth:
+      authenticator: bearertokenauth
+  # TODO: Identify any additional scrapers that are necessary and useful
+  hostmetrics:
+    scrapers:
+      cpu:
+      memory:
+      network:
+
+processors:
+  batch:
+    send_batch_size: 10000
+    timeout: 10s
+  resourcedetection:
+    detectors: [system]
+  resource:
+    attributes:
+      - key: databricks.cluster.name
+        value: \"$DB_CLUSTER_NAME\"
+        action: upsert
+      - key: databricks.cluster.id
+        value: \"$DB_CLUSTER_ID\"
+        action: upsert
+      - key: databricks.node.driver
+        value: \"$DB_IS_DRIVER\"
+        action: upsert
+
+exporters:
+  signalfx:
+    access_token: $SPLUNK_ACCESS_TOKEN
+    realm: ${SPLUNK_REALM:-us0}
+
+service:
+  extensions: [bearertokenauth]
+  pipelines:
+    metrics:
+      receivers: [hostmetrics$OPTIONAL_SPARK_RECEIVER]
+      processors: [batch, resourcedetection, resource]
+      exporters: [signalfx]
+"
+
+echo "$collector_config" > "$CONFIG_FILE"
+
+collector_service="
+[Unit]
+Description=Splunk distribution of the OpenTelemetry Collector
+StartLimitIntervalSec=0
+
+[Service]
+Type=simple
+Restart=always
+RestartSec=1
+User=root
+ExecStart=$OTEL_BINARY_FILE --config $CONFIG_FILE
+
+[Install]
+WantedBy=multi-user.target
+"
+
+echo "$collector_service" > $SERVICE_FILE
+chmod 755 $SERVICE_FILE
+
+# The collector is run as a service on the current node
+systemctl start $OTEL_BINARY_NAME
```

Comment:
```suggestion
systemctl enable "$OTEL_BINARY_NAME"
systemctl start $OTEL_BINARY_NAME
```
To get systemctl to start the collector on reboots. I don't know if this is meaningful for databricks especially if the init script mechanism will always run this script, so please ignore if this doesn't make sense.

Reply:
I did some testing, the init script is run every time a node starts up, including spinning up for the first time and restarts. Based on this, I don't think we explicitly need to enable it unless you think it's more safe to do so.